src/mbgl/gl/state.hpp

```diff
@@ -50,22 +55,43 @@ class State {
     }
 
 private:
+    template <std::size_t... I>
+    void set(std::index_sequence<I...>) {
+        T::Set(currentValue, std::get<I>(params)...);
+    }
```

Comment:
This is for passing additional arguments to some setters/getters, in particular to pass `Context` to `BindVertexArray::Set`